public/app/plugins/datasource/loki/components/LokiLabelBrowser.tsx

```diff
@@ -378,15 +390,40 @@ export class UnthemedLokiLabelBrowser extends React.Component<BrowserProps, Brow
       return <LoadingPlaceholder text="Loading labels..." />;
     }
     const styles = getStyles(theme);
+    const selector = buildSelector(this.state.labels);
+    const empty = selector === EMPTY_SELECTOR;
+
     let selectedLabels = labels.filter((label) => label.selected && label.values);
     if (searchTerm) {
-      selectedLabels = selectedLabels.map((label) => ({
+      const searchFunction = SearchFunctionMap['Fuzzy'];
```

Comment:
`SearchFunctionMap` is used internally by Slate component and has slightly different API - it matches `CompletionItems' (using `label` property) but here we need to filter `FacettableValue` (using `name` property).

I think you can use `fuzzyMatch` function (used by `SearchFunctionMap.Fuzzy`) and filter values based on the match, something like:

```typescript
if (searchTerm) {
  selectedLabels = selectedLabels.map((label) => {
    const foundValues = label.values!.filter((value) => { // label.values are guaranteed to be set by the check above
      if (value.selected) { // always include selected values
        return true;
      }
      const fuzzyMatchResult = fuzzyMatch(value.name.toLowerCase(), searchTerm.toLowerCase());
      if (fuzzyMatchResult.found) {
        value.highlightParts = fuzzyMatchResult.ranges;
        value.order = fuzzyMatchResult.distance;
        return true;
      } else {
        return false;
      }
    });
    return {
      ...label,
      values: sortBy(foundValues, (value) => (value.selected ? -Infinity : value.order)), // selected values on top
    };
  });
}
```